File: 0125-valid-palindrome/0125-valid-palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        i = 0
        j = len(s) -1

        while i < j:
            while i < j and i < len(s) and not s[i].isalnum():
                i +=1
            while j > i and j > 0 and not s[j].isalnum(): 
                j -=1
            
            if s[i].lower() != s[j].lower():
                return False
            i +=1
            j -=1
        return True
        

File: 0125-valid-palindrome/0125-valid-palindrome.cpp
#include<cctype>
class Solution {
public:
    bool isPalindrome(string s) {
        
        int l = 0;
        int r = s.size();

        while(l < r){

            while(l < r && !isalnum(s[l])){
                l++;
            }
            while(l < r && !isalnum(s[r])){
                r--;
            }

            if(std::tolower(s[l]) != std::tolower(s[r])){
                return false;
            }
            l++;
            r--;
        }

        return true;
    }
};